src/SBVonKarman.cpp

```diff
@@ -292,23 +290,40 @@ namespace galsim {
             dbg<<"sum = "<<sum<<'\n';
 
             if (_hlr == 0. && sum > thresh0) _hlr = r;
-            if (R == 0. && sum > thresh1) R = r;
         }
         _radial.finalize();
         if (_hlr == 0.)
             throw SBError("Cannot find von Karman half-light-radius.");
-        if (R == 0.) R = maxR;
         dbg<<"Finished building radial function.\n";
-        dbg<<"R = "<<R<<" arcsec\n";
         dbg<<"HLR = "<<_hlr<<" arcsec\n";
+
+        // The large r behavior of F(r) is well approximated by a power law, F ~ r^-n
+        // This affords an easier calculation of R for stepk than numerically accumulating
+        // the integral.
+        // F(r) = F1 (r/r1)^-n
+        // int_r^inf F(r) 2pi r dr = folding_threshold
+        // 2pi F1 r1^n / (n-2) f_t = R^(n-2)
+        double r1 = _radial.argMax();
+        double F1 = _radial.lookup(r1);
+        double r2 = r1 * (1-dlogr);
+        double F2 = _radial.lookup(r2);
+        dbg<<"r1,F1 = "<<r1<<','<<F1<<std::endl;
+        dbg<<"r2,F2 = "<<r2<<','<<F2<<std::endl;
+        // power law index = dlog(F)/dlog(r)
+        double n = -(std::log(F2)-std::log(F1)) / (std::log(r2)-std::log(r1));
+        // For high folding_threshold, n can come out too low (e.g. < 2), which is bad.
+        // We know the right answer when L0=inf is close to 3.67, so just use that if n < 3.
+        if (n < 3.) n = std::max(n, 3.67);
```

Comment:
I guess this one might be trickier...  Is the asymptotic slope for finite L0 < 3.67?  Or could just setting a simple floor (with no discontinuity introduced) work here too?

Reply:
Here I was less confident about what the right number is.  I didn't actually do any tests of how much the number changes for different L0.  I guess maybe I should do that investigation.  For some reasonable L0, it was empirically also pretty close to 3.67.  So maybe there is a good reason to expect it to be the same as Kolmogorov, at least over reasonable values of L0.